src/generic/stage1/json_structural_indexer.h

```diff
@@ -109,7 +109,22 @@ really_inline error_code json_structural_indexer::finish(parser &parser, size_t
     parser.structural_indexes[parser.n_structural_indexes++] = uint32_t(len);
   }
   /* make it safe to dereference one beyond this array */
-  parser.structural_indexes[parser.n_structural_indexes] = 0;
+  parser.structural_indexes[parser.n_structural_indexes] = uint32_t(len);
```

Comment:
I'm really glad you left the len there. It has the super useful property of letting you measure the maximum length of any structural by subtracting structural_indexes[n+1] - structural_indexes[n]. It also means the last character of any document in parse_many can always be found by looking at structural_indexes[last_index_in_document+1]--even if it's the last document.

Reply:
I guess you continued to increment structural_indexes, but that's fine. I'm planning to remove that, I don't think we actually need it as long as we fix up find_last_json_index and structural_indexes.past_end/at_end.